src/key.rs

```diff
@@ -27,7 +27,7 @@ use constants;
 use ffi::{self, CPtr};
 
 /// Secret 256-bit key used as `x` in an ECDSA signature
-pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
+pub struct SecretKey(pub(crate) [u8; constants::SECRET_KEY_SIZE]);
```

Comment:
Is this needed? Where do we use access the internal representation of the key?

Reply:
in `schnorrsig` module to initialize types without doing unnecessary `unwrap`s

Reply:
https://github.com/rust-bitcoin/rust-secp256k1/pull/306/commits/aa6bdaff3c7554ba018cceccf3ffcae8b40bf831#diff-d25941070387cc0b48c0525480137107aad8f631a8d8c4cd9e8d0f3a456eb6ccR465

Reply:
I think more elegant approach is to move this convertors from `schnorrsig` to `key` mod, which I did in https://github.com/rust-bitcoin/rust-secp256k1/pull/308/commits/8ee4e05e630bec8017f71669edd9e5cdc9d09cf1